enterprise/ssl-impl/src/main/java/io/crate/protocols/ssl/SslContextProviderImpl.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.protocols.ssl;
+
+import io.netty.handler.ssl.SslContext;
+import org.elasticsearch.common.settings.Settings;
+
+public class SslContextProviderImpl implements SslContextProvider {
+
+    private Settings settings = Settings.EMPTY;
+    private SslContext sslContext;
+
+    @Override
+    public void init(Settings settings) {
```

Comment:
why aren't we injecting the `Settings` into the constructor instead?

Reply:
it is not really possible, we load *SslContextProvider* with the service loader it requires to have  a no-args constructor.

Reply:
Ok so we intentionally decided against guice here, I'm ok with even if other extensions are loaded differently atm (user, license). But then I'd prefer to add the `Settings` argument to the method which is really using it: `getSslContext()`. I think adding a `init()` method which must be called first before `getSsslContext` will work is not a good pattern (at least one should assert that init() was called upfront/settings is set).

Reply:
can add an assertion to `getSslContext`, not sure what else can be done here